src/App.tsx

```diff
@@ -1,26 +1,187 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { HeaderInput } from './components/HeaderInput';
+import { Footer } from './components/Footer';
+import classNames from 'classnames';
+import { Errors, SelectedTasks, Todo } from './types/Types';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [selectedTasks, setSelectedTasks] = useState<SelectedTasks>(
+    SelectedTasks.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>();
+  const [title, setTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [deleteTodoByID, setDeleteTodoByID] = useState<number | null>();
+
```

Comment:
As a matter of code structuring: useStates => useRef, we shouldn't mix them.

Reply:
Ah, I guess my comment was misleading, my bad. You did a good job by using the useRef, I only wanted to point your attention to the hooks order in your file, it should be:

```
useStates
useRefs
useEffects
```
OR
```
useRefs
useStates
useEffects
```

So the structure is always the same, not 

```
useState
useState
useRef
useEffect
useState
```